logic/src/androidMain/kotlin/com/wire/kalium/logic/sync/WorkScheduler.kt

```diff
@@ -118,7 +136,33 @@ actual class WorkScheduler(private val context: Context, private val userId: Use
             request
         ).enqueue()
     }
+
+    override suspend fun scheduleSendingOfPersistedMessage(conversationID: ConversationId, messageUuid: String) {
+        val inputData = WrapperWorkerFactory.workDataBuilder(ScheduledMessageWorker::class, userId)
+            .scheduleMessageSending(conversationID, messageUuid)
+
+        val connectedConstraint = Constraints.Builder()
+            .setRequiredNetworkType(NetworkType.CONNECTED)
+            .build()
+
+        val request = OneTimeWorkRequest.Builder(workerClass)
+            .setConstraints(connectedConstraint)
+            .setInputData(inputData)
+            .build()
+
+        WorkManager.getInstance(context).enqueueUniqueWork(
+            WORK_NAME_PREFIX_PER_USER + userId.value,
+            ExistingWorkPolicy.APPEND,
```

Comment:
are we sure we want to append the work and not replace it ? I assume that we are could retry the sending of the same persisted message several times before we get the the internet I think in that case it would enqueue several times of sending the same message ?

Reply:
perhaps it is worth to add some integration tests with the WorkManager, dunno if you plan to do it or not in the next PR's :P

Reply:
ahhhh I see it it is W.I.P, forget it

Reply:
> are we sure we want to append the work and not replace it ? I assume that we are could retry the sending of the same persisted message several times before we get the the internet I think in that case it would enqueue several times of sending the same message ?

There's no way to try multiple times right now.
Cycle goes like this:
1. `SendWhateverMessageUseCase`
2. `MessageSender`
3. In case of failure due to bad Internet: `MessageSendingScheduler`
4. On Android, one job will be scheduled for EACH message, in order. Goes back to 2.

So it's linear. No way we spawn multiple jobs for the same message.
There's a single queue for each user: `WORK_NAME_PREFIX_PER_USER + userId.value`, and each message retry is appended in there.

If, for whatever reason we launch two workers for the same message, we can simply verify on `MessageSender` if the message is already marked as sent.

> Tests
I'm adding tests soon in this PR.